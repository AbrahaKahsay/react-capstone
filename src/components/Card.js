import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

const Card = (props) => {
  const {
    id,
    symbol,
    price,
  } = props;

  return (
    <NavLink className="card" to={`/detail/${id}`}>
      <h2>{symbol}</h2>
      <small>{id}</small>
      <p>
        $
        {price}
      </p>
    </NavLink>
  );
};

export default Card;

Card.propTypes = {
  id: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
};
