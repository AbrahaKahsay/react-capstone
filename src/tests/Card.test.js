import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Card from '../components/Card';

const coin = {
  id: 'bitcoin',
  symbol: 'BTC',
};

describe('Jest Snapshot testing suite', () => {
  test('component match with the snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Card
            key={coin.id}
            id={coin.id}
            symbol={coin.symbol}
          />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
