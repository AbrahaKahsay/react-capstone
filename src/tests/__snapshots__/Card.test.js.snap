// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite component match with the snapshot 1`] = `
<a
  className="card"
  href="/detail/bitcoin"
  onClick={[Function]}
>
  <h2>
    BTC
  </h2>
  <small>
    bitcoin
  </small>
  <p>
    $
  </p>
</a>
`;
